### DISCUSSION TITLE
Radicale IOS CardDav

### DISCUSSION THREAD

**Discussion by dm85ls** [Kozea/Radicale#1441] [Q&A general]
Address book created, By skadmin:

https://domain.ru/sa/skadmin/de9912e4-7d9b-32d2-4e3c-7b990ff0dcf8/

User dm85 created
cat /etc/radicale/rights
`[admin]
user = skadmin
collection = .*
permission = rw

[user]
user = .+
collection = .*
permission = r

[allun]
user = .*   
collection =
permission =
`
I connect the address book https://domain.ru/sa/skadmin/de9912e4-7d9b-32d2-4e3c-7b990ff0dcf8/ under the user dm85 to iphone

log:
`[1e82ab40] DEBUG: Rule 'dm85':'dm85' doesn't match 'skadmin':'.*' from section 'admin'
[1e82ab40] DEBUG: Rule 'dm85':'dm85' matches '.+':'.*' from section 'user'
[1e82ab40] WARNING: Access to principal path '/dm85/' denied by rights backend
[1e82ab40] DEBUG: Rule 'dm85':'dm85' doesn't match 'skadmin':'.*' from section 'admin'
[1e82ab40] DEBUG: Rule 'dm85':'dm85' matches '.+':'.*' from section 'user'
[1e82ab40] DEBUG: Rule 'dm85':'' doesn't match 'skadmin':'.*' from section 'admin'
[1e82ab40] DEBUG: Rule 'dm85':'' matches '.+':'.*' from section 'user'
[1e82ab40] DEBUG: Request content:
<?xml version="1.0"?>
<propfind xmlns="DAV:" xmlns:CR="urn:ietf:params:xml:ns:carddav" xmlns:CS="http://calendarserver.org/ns/">
  <prop>
    <CR:addressbook-home-set />
    <CR:directory-gateway />
    <displayname />
    <CS:email-address-set />
    <principal-collection-set />
    <principal-URL />
    <resource-id />
    <supported-report-set />
  </prop>
</propfind>

[1e82ab40] DEBUG: Rule 'dm85':'dm85' doesn't match 'skadmin':'.*' from section 'admin'
[1e82ab40] DEBUG: Rule 'dm85':'dm85' matches '.+':'.*' from section 'user'
[1e82ab40] DEBUG: Rule 'dm85':'' doesn't match 'skadmin':'.*' from section 'admin'
[1e82ab40] DEBUG: Rule 'dm85':'' matches '.+':'.*' from section 'user'
[1e82ab40] DEBUG: Response content:
The requested resource could not be found.
[1e82ab40] INFO: PROPFIND response status for '/dm85/' with depth '0' in 0.020 seconds: 404 Not Found`

I understand that he is looking for a book in dm85, and not by the full path that is specified in the settings.

How to configure to work for ios?

**Comment by return42** [comment]
> Title: Radicale IOS CardDav
...
> How to configure to work for ios?

What exactly is your aim?

**Comment by dm85ls** [comment]
You must create an address book so that only the admin can edit it, and other users only have read access.

**Comment by return42** [comment]
> You must create an address book so that only the admin can edit it, and other users only have read access.
...
> Address book created, By skadmin: https://domain.ru/sa/skadmin/de9912e4-7d9b-32d2-4e3c-7b990ff0dcf8/
...
> I connect the address book https://domain.ru/sa/skadmin/de9912e4-7d9b-32d2-4e3c-7b990ff0dcf8/ under the user dm85 to iphone
...
> I understand that he is looking for a book in dm85, and not by the full path that is specified in the settings.

you understand this (miss-) behavior?

> How to configure to work for ios?

Did you test your setup with other clients?

**Comment by dm85ls** [comment]
Works with DAVdroid and Thunderbird CardBook

Perhaps somehow it will be possible to configure ios by replacing the path with nginx? Configured according to the nginx revers proxy scheme.

**Comment by return42** [comment]
> Works with DAVdroid and Thunderbird CardBook

Sorry, I really have problems to understand you...  What works? 

If I understand you right: .. in DAVx⁵ (aka DAVdroid) you create an account for user `dm85` and with this account you can share the book with the URL from user `skadmin`?  .. may be I am wrong, but it is new for me that you can modify the URLs of user address books  in DAVx⁵ .. as far as I know: all URL of one user start with the same path e.g. for user `dm85` is `https://domain.ru/sa/dm85/...`

**Comment by dm85ls** [comment]
Yes in DAVDROID works

**Comment by return42** [comment]
> Yes in DAVDROID works

 don't know about DAVDROID please give a link and describe in detail how you setup the client

**Comment by dm85ls** [comment]
DAVdroid
I specify url https://domain.ru/sa/skadmin/de9912e4-7d9b-32d2-4e3c-7b990ff0dcf8/ 
User: dm85
Pass: ***

And it works

**Comment by return42** [comment]
Sorry, I gave up ..
